components/central-application-gateway/cmd/applicationgateway/options.go

```diff
@@ -2,53 +2,57 @@ package main
 
 import (
 	"flag"
-	"fmt"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
 type options struct {
+	apiServerURL                string
+	applicationSecretsNamespace string
 	externalAPIPort             int
+	kubeConfig                  string
+	logLevel                    *zapcore.Level
+	proxyCacheTTL               int
 	proxyPort                   int
 	proxyPortCompass            int
-	applicationSecretsNamespace string
-	requestTimeout              int
 	proxyTimeout                int
 	requestLogging              bool
-	proxyCacheTTL               int
-	kubeConfig                  string
-	apiServerURL                string
+	requestTimeout              int
 }
 
-func parseArgs() *options {
-	externalAPIPort := flag.Int("externalAPIPort", 8081, "External API port.")
-	proxyPort := flag.Int("proxyPort", 8080, "Proxy port for Kyma OS or MPS bundles with a single API definition")
-	proxyPortCompass := flag.Int("proxyPortCompass", 8082, "Proxy port for Kyma MPS.")
-	applicationSecretsNamespace := flag.String("applicationSecretsNamespace", "kyma-system", "Namespace where Application secrets used by the Application Gateway exist")
-	requestTimeout := flag.Int("requestTimeout", 1, "Timeout for services.")
-	proxyTimeout := flag.Int("proxyTimeout", 10, "Timeout for proxy call.")
-	requestLogging := flag.Bool("requestLogging", false, "Flag for logging incoming requests.")
-	proxyCacheTTL := flag.Int("proxyCacheTTL", 120, "TTL, in seconds, for proxy cache of Remote API information")
-	kubeConfig := flag.String("kubeConfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
-	apiServerURL := flag.String("apiServerURL", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+func parseArgs(log *zap.Logger) (opts options) {
```

Comment:
As I like sorting alphabetically fields in the above `options` structs, I wonder if it would make sense to adjust the order of parsing to match that struct's order